sound/soc/sof/intel/hda-ipc.c

```diff
@@ -56,6 +56,12 @@ static void hda_dsp_ipc_dsp_done(struct snd_sof_dev *sdev)
 
 int hda_dsp_ipc_send_msg(struct snd_sof_dev *sdev, struct snd_sof_ipc_msg *msg)
 {
+	if (sdev->d0_state == SOF_DSP_D0I3) {
+		snd_sof_dsp_write(sdev, HDA_DSP_BAR, CNL_DSP_REG_HIPCIDR,
+				  CNL_DSP_REG_HIPCIDR_BUSY | msg->header);
+		return 0;
+	}
```

Comment:
ditto, we should first fix the IPCs to use registers that allow all of them to use the registers. The header doesn't convey the information that's required in all cases.